db/migrate/20220825054405_fill_personal_message_enabled_groups_based_on_deprecated_settings.rb

```diff
@@ -0,0 +1,35 @@
+# frozen_string_literal: true
+
+class FillPersonalMessageEnabledGroupsBasedOnDeprecatedSettings < ActiveRecord::Migration[7.0]
+  def up
```

Comment:
@tgxworld would you mind looking over the migration too, another set of eyes is always helpful on ones like these :eyes:

Reply:
Something I noticed is that we don't seem to be disallowing write access to the old site setting in this PR. I think we need to deprecate `SiteSetting.enable_personal_messages` and `SiteSetting.min_trust_to_send_messages_raw` and create some sort of map from `SiteSetting.personal_messages_enabled_groups` to the old site settings?

Reply:
Ah very good point, that's what we did with the whitelist/allowlist stuff here, will have to do that for these settings too:

https://github.com/discourse/discourse/blob/15119b524d13f9842836aa4dcc53672569b3283c/app/models/site_setting.rb#L271-L280

Reply:
Hmm actually it doesn't really map very well, and deprecating + throwing an error when changing the setting breaks tons of specs, and I don't really want to do a fragile mapping between these two settings and the single new one. There aren't many plugins relying on this, and none of them are ours :shrug: Don't want to hold this up any longer so just adding the deprecations and being done with it.

Reply:
Never mind, found the deprecate site settings class :)